app/src/main/java/org/luxoft/sdl_core/BleSupport/BluetoothHandler.java

```diff
@@ -0,0 +1,163 @@
+    public BluetoothCentralManager central;
+    private static BluetoothHandler instance = null;
+    private BluetoothPeripheral mPeripheral = null;
+    private final Context context;
+    private final Handler handler = new Handler();
+
+    // Testing-only service with ability to notify and write
+    private static final UUID SDL_TESTER_SERVICE_UUID = UUID
+            .fromString("00001101-0000-1000-8000-00805f9b34fb");
+
+    // Testing-only characteristic with permissions to write
+    private static final UUID MOBILE_RESPONSE_CHARACTERISTIC = UUID
+            .fromString("00001103-0000-1000-8000-00805f9b34fb");
+
+    // Testing-only characteristic for notifications
+    private static final UUID MOBILE_REQUEST_CHARACTERISTIC = UUID
+            .fromString("00001102-0000-1000-8000-00805f9b34fb");
+
+
+    public static synchronized BluetoothHandler getInstance(Context context) {
+        if (instance == null) {
+            instance = new BluetoothHandler(context);
+        }
+        return instance;
+    }
+
+    private final BluetoothPeripheralCallback peripheralCallback = new BluetoothPeripheralCallback() {
+        @Override
+        public void onServicesDiscovered(BluetoothPeripheral peripheral) {
+            // Request a higher MTU, iOS always asks for 185
+            peripheral.requestMtu(185);
+
+            // Try to turn on notification
+            peripheral.setNotify(SDL_TESTER_SERVICE_UUID, MOBILE_REQUEST_CHARACTERISTIC, true);
+        }
+
+        @Override
+        public void onCharacteristicWrite(BluetoothPeripheral peripheral, byte[] value, BluetoothGattCharacteristic characteristic, GattStatus status) {
+            if (status == GattStatus.SUCCESS) {
+                Log.i(BleCentralService.TAG, "SUCCESS: Writing " + String.valueOf(value) + " to " + characteristic.getUuid());
+            } else {
+                Log.i(BleCentralService.TAG, "ERROR: Failed writing " + String.valueOf(value) + " to " + characteristic.getUuid() + " with " + status);
+            }
+        }
+
+        @Override
+        public void onCharacteristicUpdate(BluetoothPeripheral peripheral, byte[] value, BluetoothGattCharacteristic characteristic, GattStatus status) {
+            if (status != GattStatus.SUCCESS) return;
+
+            UUID characteristicUUID = characteristic.getUuid();
+            if (characteristicUUID.equals(MOBILE_REQUEST_CHARACTERISTIC)) {
+                    String msg = characteristic.getStringValue(0);
+                    Log.d(BleCentralService.TAG, "message: " + msg);
+                    Handler handler = new Handler();
+                    final BluetoothPeripheral peripheral_copy = peripheral;
+                    final String msg_copy = msg;
+                    handler.postDelayed(new Runnable() {
+                        public void run() {
+                            writeMessage(peripheral_copy, "Response to: " + msg_copy);
+                        }
+                }, 3000);
```

Comment:
@ychernysheva please add TODO comment to not forget to remove this synthetic delay in the future